beacon-chain/blockchain/process_block.go

```diff
@@ -327,6 +327,14 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []interfaces.ReadOnlySi
 	return s.saveHeadNoDB(ctx, lastB, lastBR, preState)
 }
 
+func updateNextSlotCacheOnBlock(root [32]byte, st state.BeaconState) {
```

Comment:
Why a helper function? This is only used once and I don't think anyone will be able to use this function given it has it's own slot ctx

Reply:
because it makes the whole if-else much more readable